packages/backend/src/clients/Google/GoogleDriveClient.ts

```diff
@@ -52,4 +48,47 @@ export class GoogleDriveClient {
             console.debug(`${file.name} (${file.id})`);
         });
     }
+
+    async appendToSheet(
+        refreshToken: string,
+        fileId: string,
+        csvContent: string,
+    ) {
+        if (!this.isEnabled) {
+            throw new Error('Google Drive is not enabled');
+        }
+        const authClient = await GoogleDriveClient.getCredentials(refreshToken);
+
+        const sheets = google.sheets({ version: 'v4', auth: authClient });
+
+        const tabTitle = new Date().toLocaleString().replaceAll(':', '.'); // we can't use ranges with colons in their tab ids
+
+        const sheetRows = csvContent.split('\n').map((row) => row.split(','));
+
+        await sheets.spreadsheets.batchUpdate({
+            spreadsheetId: fileId,
+            resource: {
+                requests: [
+                    {
+                        addSheet: {
+                            properties: {
+                                title: tabTitle,
+                            },
+                        },
+                    },
+                ],
+            },
+        });
+
+        await sheets.spreadsheets.values.append({
+            spreadsheetId: fileId,
+            range: `'${tabTitle}'!A:B`,
```

Comment:
if we don't want to upload this data on a new tab we just need to update this range and remove the previous `batchUpdate`